packages/freezed_annotation/lib/freezed_annotation.dart

```diff
@@ -129,6 +131,40 @@ class Freezed {
   /// ]
   /// ```
   final String? fallbackUnion;
+
+  /// Allow to deactive the maybeMap generation.
+  /// ```dart
+  /// @Freezed(shouldGenerateMaybeMap: false)
+  /// abstract class Union with _$Union {
+  ///   // ...
+  /// }
+  /// ```
+  ///
+  /// The generated code will not have the following method:
+  ///
+  /// ```dart
+  /// void main() {
+  ///   print(Union.first().maybeMap(orElse: () => null)); // Error
+  /// }
+  /// ```
+  final bool shouldGenerateMaybeMap;
```

Comment:
Looking back at this, the "shouldGenerate" is probably redundant. I think `@Freezed(maybeMap: false)` is enough

Reply:
I've changed the front facing `Freezed` class but kept `shouldGenerateMaybeMap` in `Concrete` and `Abstract` to match `shouldGenerateJson`. Let me know if you prefer to change it everywhere :)